cql3/query_result_consumer.hh

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2020 ScyllaDB
+ *
+ * Modified by ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <seastar/core/sstring.hh>
+
+#include "cql3/result_set.hh"
+#include "transport/event.hh"
+
+namespace cql3 {
+
+class query_result_consumer {
+public:
+    query_result_consumer() = default;
```

Comment:
Unneeded. A virtual destructor, however, is needed.

Reply:
done.

Reply:
This is needed after all because copy constructor is removed and that means default constructor is not generated.

Reply:
To be fair, virtual destructor wasn't needed as well because consumers are created and destroyed using their concrete types and just passed down using reference to interface/abstract class. We added virtual destructor though because it's safer.